libraries/llvm/types.c

```diff
@@ -0,0 +1,20 @@
+#ifndef EFFEKT_TYPES_C
+#define EFFEKT_TYPES_C
+
+
+typedef int64_t Int;
+typedef double Double;
+
+struct Pos {
+    uint64_t tag; // type-local tag
+    void *obj; // pointer into the heap
+};
+
+static const struct Pos Unit = (struct Pos) { .tag = 0, .obj = NULL, };
+static const struct Pos BooleanFalse = (struct Pos) { .tag = 0, .obj = NULL, };
+static const struct Pos BooleanTrue = (struct Pos) { .tag = 1, .obj = NULL, };
+
+typedef struct Pos String;
```

Comment:
Create a separate struct type for strings, (same as on the llvm side), and call the fields, hmmm, `len` and `raw` or something.

Reply:
On the LLVM side, there is no `%String` type no more, since bit-reinterpretations would need to arduously be explicitly performed.

Reply:
Ok, yes, you are right, the fact that strings have the same representation as data types is a feature.